catroid/src/androidTest/java/org/catrobat/catroid/test/content/bricks/BrickPermissionTest.java

```diff
@@ -0,0 +1,175 @@
+				{"CameraBrick", new Brick[]{new CameraBrick()}, new String[]{CAMERA}},
+				{"LegoNxtMotorMoveBrick", new Brick[]{new LegoNxtMotorMoveBrick()}, new String[]{BLUETOOTH_ADMIN, BLUETOOTH}},
+				{"CameraBrick + LegoNxtMotorTurnAngleBrick", new Brick[]{new CameraBrick(), new LegoNxtMotorTurnAngleBrick()}, new String[]{CAMERA, BLUETOOTH_ADMIN, BLUETOOTH}},
+				{"AskSpeechBrick", new Brick[]{new AskSpeechBrick()}, new String[]{RECORD_AUDIO}},
+				{"WhenGamepadButtonBrick", new Brick[]{new WhenGamepadButtonBrick(new WhenGamepadButtonScript())}, new String[]{CHANGE_WIFI_MULTICAST_STATE, CHANGE_WIFI_STATE, ACCESS_WIFI_STATE}},
+				{"WhenNfcBrick", new Brick[]{new WhenNfcBrick()}, new String[]{NFC}},
+				{"WhenNfcBrick + GPS", new Brick[]{new WhenNfcBrick(), brickWithGPS}, new String[]{NFC, ACCESS_FINE_LOCATION, ACCESS_COARSE_LOCATION}},
+				{"Brick With GPS Formula", new Brick[]{brickWithGPS}, new String[]{ACCESS_FINE_LOCATION, ACCESS_COARSE_LOCATION}}
+		});
+	}
+
+	@Parameterized.Parameter
+	public String name;
+
+	@Parameterized.Parameter(1)
+	public Brick[] bricks;
+
+	@Parameterized.Parameter(2)
+	public String[] expectedPermission;
+
+	Script script;
+	@Before
+	public void setUp() {
+		Project project = new Project();
+		Scene scene = new Scene();
+		project.addScene(scene);
+		Sprite sprite = new Sprite();
+		scene.addSprite(sprite);
+		script = new StartScript();
+		sprite.addScript(script);
+		ProjectManager.getInstance().setCurrentProject(project);
+	}
+
+	@Test
+	public void testGetProjectRuntimePermissionList() {
+		script.getBrickList().addAll(Arrays.asList(bricks));
+		assertRuntimePermissions();
+	}
+
+	@Test
+	public void testRuntimePermissionInsideIf() {
+		IfLogicBeginBrick ifBrick = new IfLogicBeginBrick();
+		for (Brick brick : bricks) {
+			ifBrick.addBrickToIfBranch(brick);
+		}
+		script.addBrick(ifBrick);
+		assertRuntimePermissions();
+	}
+
+	@Test
+	public void testRuntimePermissionInsideElse() {
+		IfLogicBeginBrick ifBrick = new IfLogicBeginBrick();
+		for (Brick brick : bricks) {
+			ifBrick.addBrickToElseBranch(brick);
+		}
+		script.addBrick(ifBrick);
+		assertRuntimePermissions();
+	}
+
+	@Test
+	public void testRuntimePermissionInsideForever() {
+		ForeverBrick foreverBrick = new ForeverBrick();
+		for (Brick brick : bricks) {
+			foreverBrick.addBrick(brick);
+		}
+		script.addBrick(foreverBrick);
+		assertRuntimePermissions();
+	}
+
+	@Test
+	public void testRuntimePermissionInsideRepeat() {
+		RepeatBrick repeatBrick = new RepeatBrick();
+		for (Brick brick : bricks) {
+			repeatBrick.addBrick(brick);
+		}
+		script.addBrick(repeatBrick);
+		assertRuntimePermissions();
+	}
+
+	@Test
+	public void testDoubleNestedPermission() {
+		RepeatBrick repeatBrick0 = new RepeatBrick();
+		RepeatBrick repeatBrick1 = new RepeatBrick();
+		for (Brick brick : bricks) {
+			repeatBrick1.addBrick(brick);
+		}
+		repeatBrick0.addBrick(repeatBrick1);
+
+		script.addBrick(repeatBrick0);
+		assertRuntimePermissions();
+	}
+
```

Comment:
could you add a test, where not all bricks are added, thus not all permissions are needed. It would be cool to check, if only the needed permissions are requested, because we would't want the app to request all permissions all the time, only the needed ones.